CHANGELOG.rst

```diff
@@ -1,63 +1,88 @@
-  cumbersome and rarely used. Use ``qtbot.stop()`` (added in 1.1.1) instead.
+  cumbersome and rarely used. Use ``qtbot.stop()`` (added in 1.1.1) instead
+  (`#306`_). Thanks `@The-Compiler`_ for the PR.
 - The old ``SignalTimeoutError`` exception alias is now removed, as it was renamed to
-  ``TimeoutError`` in 2.1.
+  ``TimeoutError`` in 2.1 (`#306`_). Thanks `@The-Compiler`_ for the PR.
 - The old ``qt_wait_signal_raising`` option is now removed, as it was renamed to
-  ``qt_default_raising`` in 3.1.
+  ``qt_default_raising`` in 3.1 (`#306`_). Thanks `@The-Compiler`_ for the PR.
 - ``qtbot.waitSignal`` and ``waitSignals`` (as well as their PEP-8 aliases)
   supported passing ``None`` as signal, making them wait for the given timeout
-  instead. This is not supported anymore, use ``qtbot.wait(ms)`` instead.
+  instead. This is not supported anymore, use ``qtbot.wait(ms)`` instead
+  (`#306`_). Thanks `@The-Compiler`_ for the PR.
 - Various classes are now not importable from ``pytestqt.plugin`` anymore, and
   should instead be imported from the module they're residing in since the 1.6.0
-  release:
+  release (`#306`_):
 
   * ``pytestqt.plugin.QtBot`` -> ``pytestqt.qtbot.QtBot``
   * ``pytestqt.plugin.SignalBlocker`` -> ``pytestqt.wait_signal.SignalBlocker``
   * ``pytestqt.plugin.MultiSignalBlocker`` -> ``pytestqt.wait_signal.MultiSignalBlocker``
   * ``pytestqt.plugin.Record`` -> ``pytestqt.logging.Record``
-  * ``pytestqt.plugin.capture_exceptions`` -> ``pytestqt.exceptions.capture_exceptions`` (but consider using ``qtbot.capture_exceptions`` instead)
-  * ``pytestqt.plugin.format_captured_exceptions`` -> ``pytestqt.exceptions.format_captured_exceptions``
+  * ``pytestqt.plugin.capture_exceptions`` ->
+    ``pytestqt.exceptions.capture_exceptions`` (but consider using
+    ``qtbot.capture_exceptions`` instead)
+  * ``pytestqt.plugin.format_captured_exceptions`` ->
+    ``pytestqt.exceptions.format_captured_exceptions``
+- The ``qt_api.extract_from_variant`` and ``qt_api.make_variant`` functions
+  (which were never intended for public usage) as well as all class aliases
+  (such as ``qt_api.QWidget`` or ``qt_api.QEvent``, among others) are now
+  removed. Thanks `@The-Compiler`_ for the PR.
 - The default timeouts for ``qtbot.waitSignal``, ``waitSignals``, ``waitUntil``
-  and ``waitCallback`` have been raised from 1s to 5s. This makes them in line
-  with the existing ``waitActive`` and ``waitExposed`` timeouts, as well as the
-  default timeout used by Qt's ``QSignalSpy::wait``. To get the old behavior
-  back, explicitly pass ``timeout=1000`` to those functions.
+  and ``waitCallback``, ``waitActive`` and ``waitExposed`` have been raised from
+  1s to 5s. This makes them in line the default timeout used by Qt's underlying
+  methods such as ``QSignalSpy::wait``. To get the old behavior back, explicitly
+  pass ``timeout=1000`` to those functions (`#306`_). Thanks `@The-Compiler`_
```

Comment:
While reading this I realized `waitSignal` (and friends) have this signature:

```python
def waitSignal(self, signal, timeout=5000, raising=None, check_params_cb=None):
```

While IMHO this would be better:

```python
def waitSignal(self, signal, *, timeout=5000, raising=None, check_params_cb=None):
```

(Of course this is an artifact from Python 2 support).

Don't want to pile this on for 4.0, just commenting to see what you think.

Reply:
Let's do this for 4.0! See #366. Makes total sense, and should be an easy change.